src/js/media-display-gesture-receiver.js

```diff
@@ -0,0 +1,142 @@
+import { MediaUIAttributes, MediaUIEvents, PointerTypes } from './constants.js';
+import { defineCustomElement } from './utils/defineCustomElement.js';
+import {
+  Window as window,
+  Document as document,
+} from './utils/server-safe-globals.js';
+
+const template = document.createElement('template');
+
+template.innerHTML = `
+<style>
+  :host {
+    display: inline-block;
+    width: auto;
+    height: auto;
+    box-sizing: border-box;
+
+    pointer-events: auto;
+  }
+</style>
+`;
+
+class MediaDisplayGestureReceiver extends window.HTMLElement {
+  // NOTE: Currently "baking in" actions + attrs until we come up with
+  // a more robust architecture (CJP)
+  static get observedAttributes() {
+    return [MediaUIAttributes.MEDIA_CONTROLLER, MediaUIAttributes.MEDIA_PAUSED];
+  }
+
+  constructor(options = {}) {
+    super();
+
+    const shadow = this.attachShadow({ mode: 'open' });
+
+    const buttonHTML = template.content.cloneNode(true);
+    this.nativeEl = buttonHTML;
+
+    // Slots
+    let slotTemplate = options.slotTemplate;
+
+    if (!slotTemplate) {
+      slotTemplate = document.createElement('template');
+      slotTemplate.innerHTML = `<slot>${options.defaultContent || ''}</slot>`;
+    }
+
+    this.nativeEl.appendChild(slotTemplate.content.cloneNode(true));
+
+    shadow.appendChild(buttonHTML);
+
+    this._pointerType = undefined;
+    const pointerDownHandler = (e) => {
+      // Since not all browsers have updated to be spec compliant, where 'click' events should be PointerEvents,
+      // we can use use 'pointerdown' to reliably determine the pointer type. (CJP).
+      this._pointerType = e.pointerType;
+    };
+
+    this.addEventListener('pointerdown', pointerDownHandler);
+
+    this.addEventListener('click', (event) => {
+      if (!event.pointerType) {
+        event.pointerType = this._pointerType;
+      }
+      const { pointerType } = event;
+      // NOTE: While there are cases where we may have a stale this._pointerType,
+      // we're guaranteed that the most recent this._pointerType will correspond
+      // to the current click event definitionally. As such, this clearing is technically
+      // unnecessary (CJP)
+      this._pointerType = undefined;
+
+      // NOTE: Longer term, we'll likely want to delay this to support double click/double tap (CJP)
+      if (pointerType === PointerTypes.TOUCH) {
+        this.handleTap(event);
+        return;
+      } else if (pointerType === PointerTypes.MOUSE) {
+        this.handleMouseClick(event);
+        return;
+      } else if (pointerType === PointerTypes.PEN) {
+        this.handlePenTap(event);
+        return;
```

Comment:
might be a little early for this use case, are we going to do special things for touch pens?

Reply:
just wanted to make sure it's accounted for somewhere since it's the only other well-defined pointerType, but happy to remove it for now if you're concerned?

Reply:
concern is YAGNI

Reply:
sold. Removing.